samcli/lib/package/packageable_resources.py

```diff
@@ -178,7 +178,7 @@ def get_property_value(self, resource_dict):
             return {"Bucket": None, "Key": None}
 
         resource_path = jmespath.search(self.PROPERTY_NAME, resource_dict)
-        if resource_path:
+        if resource_path and isinstance(resource_path, str):
```

Comment:
does this change to cove if the S3 information is added as Ref intrinsic function?

Reply:
Yes that is correct.

Reply:
- shall you add a comment on that, and also can you explain in this case what will be the expected behaviour (shall SAM CLI only delete the stack, and skip deleting the artifact in S3)
- Also add TODO message to fix this
- One more thing the Ref intrinsic can be used in the Image Resources as well, so I think you need to do the same in its class.

Reply:
That makes sense, I will make those changes.